### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by 1nsider21**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -46,7 +46,25 @@
  * @param {Object[]} transforms
  */
 function transformState(state, transforms) {
-  // write code here
+  for (const value of transforms) {
+    if (value.operation === 'addProperties') {
+      for (const i in value.properties) {
+        state[i] = value.properties[i];
```
Comment (tykhan) [inline src/transformState.js:52]: щоб не робити ще один цикл, можна скористатись Object.assign

**Review by tykhan — APPROVED** [review]

File: src/transformState.js (hunk 1 above)
Comment (vsm2k17) [inline src/transformState.js:52]:
```suggestion

        state = {
        ...state,
        ...properties,
```

File: src/transformState.js [hunk 2]
```diff
@@ -46,7 +46,25 @@
...
+      }
+    } else if (value.operation === 'removeProperties') {
+      for (const i of value.properties) {
```
Comment (vsm2k17) [inline src/transformState.js:55]: "i" - that bad name, when you use "for of" loop, better to use something more meaningful

File: src/transformState.js [hunk 3]
```diff
@@ -46,7 +46,25 @@
...
+        if (state[i]) {
+          delete state[i];
+        }
+      }
+    } else if (value.operation === 'clear') {
+      for (const i in state) {
```
Comment (vsm2k17) [inline src/transformState.js:61]: "i" that's a bad name

**Review by mykola-yarchuk — APPROVED** [review]
consider of using switch construction

**Pull request closed without merging**